File: app/src/main/java/feelings/guide/question/QuestionService.java
package feelings.guide.question;

import android.content.Context;
import android.database.Cursor;
import android.util.Log;

import feelings.guide.answer.AnswerStore;

public class QuestionService {

    private static final String TAG = QuestionService.class.getSimpleName();

    public static final long FEELINGS_ID = 1;

    public static String getQuestionText(Context context, long questionId) {
        Question question = QuestionStore.getById(context, questionId);
        return question != null
                ? question.getText()
                : "Couldn't get question text";
    }

    public static long createQuestion(Context context, Question question) {
        return QuestionStore.createQuestion(context, question);
    }

    public static boolean updateQuestion(Context context, Question question) {
        if (question.getId() == FEELINGS_ID) {
            Log.e(TAG, "updateQuestion: attempt to update feelings question!");
            return false;
        }
        return QuestionStore.updateQuestion(context, question);
    }

    public static boolean deleteQuestion(Context context, long questionId) {
        if (questionId == FEELINGS_ID) {
            Log.e(TAG, "deleteQuestion: attempt to delete feelings question!");
            return false;
        }
        return QuestionStore.deleteQuestion(context, questionId);
    }

    public static boolean hideQuestion(Context context, long questionId) {
        if (questionId == FEELINGS_ID) {
            Log.e(TAG, "hideQuestion: attempt to hide feelings question!");
            return false;
        }
        return QuestionStore.hideQuestion(context, questionId);
    }

    public static void restoreHidden(Context context) {
        QuestionStore.restoreHidden(context);
    }

    public static Cursor getAllQuestions(Context context) {
        return QuestionStore.getAll(context);
    }

    public static Question getById(Context context, long questionId) {
        Question question = QuestionStore.getById(context, questionId);
        if (question == null) {
            Log.e(TAG, "getById: couldn't find question by id=" + questionId);
        }
        return question;
    }

    public static boolean changeLanguage(Context context) {
        return QuestionStore.changeLanguage(context);
    }

    public static boolean hasAnswers(Context context, long questionId) {
        return AnswerStore.hasAnswers(context, questionId);
    }
}

File: app/src/main/kotlin/feelings/guide/question/QuestionService.kt
package feelings.guide.question

import android.content.Context
import android.database.Cursor
import android.util.Log

import feelings.guide.answer.AnswerStore

object QuestionService {

    private val TAG = QuestionService::class.java.simpleName

    const val FEELINGS_ID: Long = 1

    fun getQuestionText(context: Context, questionId: Long): String? {
        val question = QuestionStore.getById(context, questionId)
        return if (question != null)
            question.text
        else
            "Couldn't get question text"
    }

    fun createQuestion(context: Context, question: Question): Long {
        return QuestionStore.createQuestion(context, question)
    }

    fun updateQuestion(context: Context, question: Question): Boolean {
        if (question.id == FEELINGS_ID) {
            Log.e(TAG, "updateQuestion: attempt to update feelings question!")
            return false
        }
        return QuestionStore.updateQuestion(context, question)
    }

    fun deleteQuestion(context: Context, questionId: Long): Boolean {
        if (questionId == FEELINGS_ID) {
            Log.e(TAG, "deleteQuestion: attempt to delete feelings question!")
            return false
        }
        return QuestionStore.deleteQuestion(context, questionId)
    }

    fun hideQuestion(context: Context, questionId: Long): Boolean {
        if (questionId == FEELINGS_ID) {
            Log.e(TAG, "hideQuestion: attempt to hide feelings question!")
            return false
        }
        return QuestionStore.hideQuestion(context, questionId)
    }

    fun restoreHidden(context: Context) {
        QuestionStore.restoreHidden(context)
    }

    fun getAllQuestions(context: Context): Cursor {
        return QuestionStore.getAllVisible(context)
    }

    fun getById(context: Context, questionId: Long): Question? {
        val question = QuestionStore.getById(context, questionId)
        if (question == null) {
            Log.e(TAG, "getById: couldn't find question by id=$questionId")
        }
        return question
    }

    fun changeLanguage(context: Context) {
        QuestionStore.changeLanguage(context)
    }

    fun hasAnswers(context: Context, questionId: Long): Boolean {
        return AnswerStore.hasAnswers(context, questionId)
    }
}
